test-app/app/components/element-receiver.ts

```diff
@@ -0,0 +1,14 @@
+import Component from '@glimmer/component';
+import type { ElementSignature, ElementFromTagName } from 'ember-element-helper';
+
+interface ElementReceiverSignature<T extends string> {
+  Element: ElementFromTagName<T>;
+  Args: {
+    tag: ElementSignature<T>['Return'];
+  };
+  Blocks: {
+    default: [];
+  }
+}
```

Comment:
This is my test type, but they appear not to work perfectly. Here is the glint error message:

```
app/components/element-receiver.hbs:2:3 - error TS2345: Argument of type 'NonNullable<ElementFromTagName<T>> extends never ? unknown : ElementFromTagName<T>' is not assignable to parameter of type 'Element'.
  Type 'unknown' is not assignable to type 'Element'.

2   <Tag id="content" ...attributes>{{yield}}</Tag>
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

app/components/element-receiver.hbs:2:21 - error TS2345: Argument of type 'NonNullable<ElementFromTagName<T>> extends never ? unknown : ElementFromTagName<T>' is not assignable to parameter of type 'Element'.
  Type 'unknown' is not assignable to type 'Element'.

2   <Tag id="content" ...attributes>{{yield}}</Tag>
```